opentelemetry-sdk/src/opentelemetry/sdk/_shared_internal/__init__.py

```diff
@@ -180,16 +183,36 @@ def emit(self, data: Telemetry) -> None:
         if len(self._queue) >= self._max_export_batch_size:
             self._worker_awaken.set()
 
-    def shutdown(self):
+    def shutdown(self, timeout_millis: int = 30000):
         if self._shutdown:
             return
-        # Prevents emit and force_flush from further calling export.
+        # Causes emit to reject telemetry and makes force_flush a no-op.
         self._shutdown = True
-        # Interrupts sleep in the worker, if it's sleeping.
+        # Interrupts sleep in the worker if it's sleeping.
         self._worker_awaken.set()
-        # Main worker loop should exit after one final export call with flush all strategy.
-        self._worker_thread.join()
-        self._exporter.shutdown()
+        # Wait a tiny bit for the worker thread to wake and call export for a final time.
+        time.sleep(0.1)
+        num_sleeps = 10
+        for _ in range(num_sleeps):
+            # If export is not being called, we can shutdown.
+            if not self._export_lock.locked():
+                break
+            time.sleep(timeout_millis / 1000 / num_sleeps)
```

Comment:
I like the way you did it in the OTLP exporter where you add an Event, is it possible to do something similar here instead of this sleeping?